lib/text/ui_text_displayer.js

```diff
@@ -189,6 +189,133 @@ shaka.text.UITextDisplayer = class {
     }
   }
 
+  /**
+   * Returns tokens describing start/end tags and string content from a cue.
+   *
+   * @param {string} str
+   * @return {!Array<{type: string, value: string}>}
+   * @private
+   */
+  tokenizeCueString_(str) {
+    if (!str) {
+      return [];
+    }
+
+    const tokens = [];
+
+    let currentValue = '';
+    for (let i = 0; i < str.length; i++) {
+      if (str[i] === '<' && str[i+1] !== ' ') {
```

Comment:
It's a fair assumption that a '<' won't appear at the very end of the string, but I think it's still worth adding a check to see if `i+1 < str.length` before accessing it as an index.

Reply:
I changed it!